custom_components/samsungtv_tizen/logo.py

```diff
@@ -0,0 +1,209 @@
+                    self._media_image_base_url = None
+                else:
+                    self._media_image_base_url = "{}export/{}".format(
+                        LOGO_BASE_URL, logo_option
+                    )
+                self._logo_option = logo_option
+            else:
+                _LOGGER.warning(
+                    "Unrecognized value '%s' for 'Display logos' option. Using default value.",
+                    logo_option,
+                )
+                self._logo_option = LOGO_OPTION_DEFAULT[1]
+                self._media_image_base_url = "{}export/{}".format(
+                    LOGO_BASE_URL, LOGO_OPTION_DEFAULT[1]
+                )
+
+    async def _async_ensure_latest_path_file(self):
+        """ Does check if logo paths file exists and if it does - is it out of date or not. """
+        if self._media_image_base_url is not None and (
+            self._last_check is None
+            or self._last_check
+            < (
+                datetime.now().astimezone()
+                - timedelta(days=LOGO_FILE_DAYS_BEFORE_UPDATE)
+            )
+        ):
+            if os.path.isfile(self._logo_file_download_path):
+                file_date = datetime.utcfromtimestamp(
+                    os.path.getmtime(self._logo_file_download_path)
+                ).astimezone()
+                if file_date < (
+                    datetime.now().astimezone()
+                    - timedelta(days=LOGO_FILE_DAYS_BEFORE_UPDATE)
+                ):
+                    try:
+                        async with self._session.head(
+                            LOGO_BASE_URL + "logo_paths.json"
+                        ) as response:
+                            url_date = datetime.strptime(
+                                response.headers.get("Last-Modified"),
+                                "%a, %d %b %Y %X %Z",
+                            ).astimezone()
+                            if url_date > file_date:
+                                self._last_check = datetime.now().astimezone()
+                                await self._download_latest_path_file()
+                    except:
+                        _LOGGER.warning(
+                            "Not able to download latest paths file for logos. Using out-of-date paths file."
+                        )
+            else:
+                self._last_check = datetime.now().astimezone()
+                await self._download_latest_path_file()
+
+    async def _download_latest_path_file(self):
+        try:
+            async with self._session.get(LOGO_BASE_URL + "logo_paths.json") as response:
+                response = (await response.read()).decode("utf-8")
+                if response:
+                    async with aiofiles.open(
+                        self._logo_file_download_path, mode="w+", encoding="utf-8"
+                    ) as paths_file:
+                        await paths_file.write(response)
+        except:
```

Comment:
It is possible to use a more specific exception?

Reply:
Done! Made a distinction in errors by not being able to connect to download and writing the file to disk.